server/node/create_sequence.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package node
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+	vitess "github.com/dolthub/vitess/go/vt/sqlparser"
+
+	"github.com/dolthub/doltgresql/core"
+	"github.com/dolthub/doltgresql/core/sequences"
+)
+
+// CreateSequence handles the CREATE SEQUENCE statement, along with SERIAL type definitions.
+type CreateSequence struct {
+	schema      string
+	ifNotExists bool
+	sequence    *sequences.Sequence
+}
+
+var _ sql.ExecSourceRel = (*CreateSequence)(nil)
+var _ vitess.Injectable = (*CreateSequence)(nil)
+
+// NewCreateSequence returns a new *CreateSequence.
+func NewCreateSequence(ifNotExists bool, schema string, sequence *sequences.Sequence) *CreateSequence {
+	return &CreateSequence{
+		schema:      schema,
+		ifNotExists: ifNotExists,
+		sequence:    sequence,
+	}
+}
+
+// CheckPrivileges implements the interface sql.ExecSourceRel.
+func (c *CreateSequence) CheckPrivileges(ctx *sql.Context, opChecker sql.PrivilegedOperationChecker) bool {
+	// TODO: implement privilege checking
+	return true
+}
+
+// Children implements the interface sql.ExecSourceRel.
+func (c *CreateSequence) Children() []sql.Node {
+	return nil
+}
+
+// IsReadOnly implements the interface sql.ExecSourceRel.
+func (c *CreateSequence) IsReadOnly() bool {
+	return false
+}
+
+// Resolved implements the interface sql.ExecSourceRel.
+func (c *CreateSequence) Resolved() bool {
+	return true
+}
+
+// RowIter implements the interface sql.ExecSourceRel.
+func (c *CreateSequence) RowIter(ctx *sql.Context, r sql.Row) (sql.RowIter, error) {
+	if strings.HasPrefix(strings.ToLower(c.sequence.Name), "dolt") {
+		return nil, fmt.Errorf("sequences cannot be prefixed with 'dolt'")
+	}
+	schema := c.schema
+	if len(c.schema) == 0 {
+		schema = core.GetCurrentSchema(ctx)
```

Comment:
Need a failure case here: currentSchema will be empty if the search_path is empty